### PR TITLE
Prompt for email on register flow #1225

### LINKED ISSUE #1225: Send user to login source interstitial when registering
The user should go to the login source interstitial when registering so that they login with their AMO account before realising they have a sync account under a different email.

### PR DESCRIPTION
**Pull request by mstriemer**
fxa-register-prompt-email-two mov

This fixes #1225.

### CODE REVIEW

File: apps/amo/helpers.py [hunk 1]
```diff
@@ -237,18 +237,27 @@ def page_name(app=None):
         return _('Add-ons')
 
 
-@register.function
-@jinja2.contextfunction
-def login_link(context):
-    next = context['request'].path
+def link_with_to_param(context, base):
```
Comment (kumar303) [inline apps/amo/helpers.py]: `link_with_final_destination()` would read less awkwardly to me

File: apps/amo/helpers.py [hunk 2]
```diff
@@ -237,18 +237,27 @@ def page_name(app=None):
...
+    next_path = context['request'].path
 
     qs = context['request'].GET.urlencode()
 
     if qs:
-        next += '?' + qs
+        next_path += '?' + qs
```
Comment (kumar303) [inline apps/amo/helpers.py]: Should we capture hashes too? I guess we don't really use them but someone might in the future.

Comment (kumar303) [inline apps/amo/helpers.py]: hmm, maybe Django doesn't even see those so nevermind

**Comment by kumar303** [comment]
r+wc

File: apps/amo/helpers.py (hunk 1 above)
Comment (mstriemer) [inline apps/amo/helpers.py]: Sounds scary. I'll change it.

**Pull request merged**